content/hardware/05.pro-solutions/solutions-and-kits/portenta-machine-control/tutorials/pmc-opta-modbus-tcp/content.md

```diff
@@ -0,0 +1,288 @@
+#include <PortentaEthernet.h>
+arduino::EthernetClass eth(&m_netInterface);
+
+void setup()
+{
+	// Configure static IP address
+	IPAddress ip(10, 0, 0, 157);    // PMC IP address
+	IPAddress dns(10, 0, 0, 1);     // gateway IP address
+	IPAddress gateway(10, 0, 0, 1); // gateway IP address
+	IPAddress subnet(255, 255, 255, 0); 
+	// If cable is not connected this will block the start of PLC with about 60s of timeout!
+	eth.begin(ip, dns, gateway, subnet);
+
+}
+```
+![Network settings for Modbus TCP](assets/ip-setup.png)
+
+Now, create the variable that will be shared with the temperature sensor data in the network. For this, we go to **status variables** and click on **Add**. Configure it as follows:
+
+- Name: `temp_send`
+- Address: `25000`
+- PLC type: `REAL`
+
+![Temperature variable setup](assets/var-setup.png)
+
+Next, go to **Temperature probes** and select the sensor type, for this tutorial, enable the **thermocouple** connected to the first channel by setting it to true.
+
+![Temperature probe configuration](assets/probe-setup.png)
+
+Finally, go to the **main program** in **project** and match the temperature variable with the temperature sensor lectures as follows:
+
+```
+temp_send := sysTempProbes[0].temperature; 
+```
+
+![Main program for the server](assets/main-code.png)
+
+To check if everything is okay, click on compile, and if no error is shown, you can upload the code to your Portenta Machine Control. 
+
+![Project compiled with no errors](assets/compile.png)
+
+Once uploaded, click on the **Connect** button again. Now, you can monitor the **temp_send** variable in the **Watch** window dragging and dropping it from the __Global shared__ variables in the project tab.
+
+You should see the temperature value measured by the sensor.
+
+![Live temperature measurement on the PMC](assets/temp-check.gif)
+
+### Opta Micro PLC Setup
+
+Now the server is configured, create a new project, this time for the Opta™ micro PLC that will be the Client or Master.
+
+![New project for the Opta™](assets/new-project-opta.png)
+
+Upload the runtime for Opta™ by selecting its serial port and clicking on the **Download** button as before.
+
+![Uploading the runtime to the Opta™](assets/runtime-opta.png)
+
+Once the runtime is flashed, with your Opta™ connected to your router, search for its IP address on the router configurations.
+
+On the PLC IDE, navigate to **On-line > Set up communication**, activate and then open the **ModbusTCP** properties. Add the Opta™ IP address, then click "OK". 
+
+![Modbus TCP connection](assets/modbus-prog-opta.png)
+![Modbus TCP IP setup](assets/modbus-ip-opta.png)
+
+Now, in the upper left corner, click on the **Connect** button and wait for the base program to be uploaded. A green **Connected** flag should appear in the lower right corner if everything goes well.
+
+![Connecting the board](assets/connected-opta.png)
+
+***The Opta™ doesn’t need any license activation to be used with the PLC IDE***
```

Comment:
Are we sure about this? there are still some steps to do with pre-licensed products like Opta:

https://docs.arduino.cc/tutorials/portenta-machine-control/plc-ide-setup-license/?queryID=undefined#7-license-activation-with-pre-licensed-products-opta

Please check and update the tutorial accordingly

Reply:
Noted! Thanks